input/input-remarkable.h

```diff
@@ -0,0 +1,26 @@
+/*
+    KOReader: Remarkable input abstraction for Lua
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef _KO_INPUT_REMARKABLE_H
+#define _KO_INPUT_REMARKABLE_H
+
+static void generateFakeEvent(int pipefd[2]) {
+	// TODO anything required here? Sony PRS port monitors battery charge etc
```

Comment:
You can generate fake events that will be forwarded to the frontend as normal input events. There are a few defined in https://github.com/koreader/koreader-base/blob/master/input/input.c#L36-L41 but you can define more.

Normally is used to notify the program hardware changes, and we use libue to get that info from udev.

Once you have your fake events ready you can react to them in UIManager, but first do note that names are slightly different in the frontend. See how they're named in https://github.com/koreader/koreader/blob/f1f75c5cb0fa063c35f49a84b2e025f1d292ea01/frontend/device/input.lua#L176